enhancements/results-forwarding.md

```diff
@@ -0,0 +1,379 @@
+- worker 
+- master 
+scanTolerations: 
+  default:
+  - operator: Exists
+  schedule: 0 1 * * * 
+```
+
+To forward results, we'd implement a `grpc` provider. A sample would look as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha2
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+generateResultCRDs: false # no CRD-creator container will be provisioned
+storeEvidenceInPVC: false # no PVC store container will be provisioned
+evidenceStorage: null # No storage options are needed
+resultForwarding:
+  provider: grpc
+  grpc:
+    version: v1  # we'll start at v1
+    endpoint: <URL to GRPC service (could be in-cluster or outside of it)>
+    tls:
+      dynamic: 
+      caSecretName: <ref to CA cert/key pair for TLS> . # This is mandatory
+      serverSecretName: <ref to server cert/key pair secret for mTLS>
+      clientSecretName: <ref to client cert/key pair secret for mTLS> # Mandatory if mtls authentication is used
+    authentication:
+      type: <mtls or token>
+      token:  # May be nil if mtls is chosen
+        tokenSecretName: <ref to secret containing token>
+    extraMetadata:  # defined as a map[string]string
+      cluster_name: foo
+      cluster_type: bar
+      randomKey: random value
+...
+```
+
+Note that `generateResultCRDs` is set to `false`, as well as `storeEvidenceInPVC`
+
+```yaml
+evidenceForwarding:
+  provider: s3
+  s3:
+    bucketName: <name of the s3 bucket>
+    objectPrefix:
+    accessKeyID: 
+```
+
+Potential evidence forwarding providers could be implemented with a similar scheme. e.g.
+a potential `http` provider could look as follows:
+
+```yaml
+evidenceForwarding:
+  provider: http
+  http:
+    version: v1
+    endpoint: <URL of the receiving endpoint>
+    tls:
+      caSecretName: <ref to CA cert/key pair for TLS>
+      serverSecretName: <ref to server cert/key pair secret for mTLS>
+      clientSecretName: <ref to client cert/key pair secret for mTLS>
+```
+
+Note that the current implementation proposal will include the `s3` and `pvc`
+providers only.
+
+### Implementation Details/Notes/Constraints
+
+A First step in implementing the forwarding proposal would be to work with the existing
+`v1alpha1` `ScanSetting` object. All changes would need to be done in how the Compliance
+Operator effectuates scans, and thus we'd ensure that backwards compatibility is
+taken into account. A new `crd-creator` pod would be introduced to receive the results
+from the `aggregator` and ensure that the known `ComplianceCheckResult` and
+`ComplianceRemediation` objects are created.
+
+A subsequent step would be introduce the `v1alpha2` `ScanSetting` object, which would
+contain all the tunables needed for the forwarding architecture.
+
+A webhook would be introduced to ensure compatibility between the `v1alpha1` objects
+and the new `v1alpha2` objects.
+
+
+### Risks and Mitigations
+
+`v1alpha1` objects would be easily translated to `v1alpha2` objects, but not the
```

Comment:
This is why I'd suggest that we make generateResultCRDs and storeEvidenceInPVC both default to true, with empty forwarding configs. This would make the transition to v1alpha2 automatic. Once you're updated to v1alpha2 you can then configure forwarding and turn off CRD/PVC as desired.

Reply:
I thought I had already written that they default to true (as is the configuration today).

Reply:
It would have been a terrible decision to default to anything else as that would break existing deployments, which is not the intention.